.github/workflows/main.yml

```diff
@@ -0,0 +1,67 @@
+name: Create new release
+
+on:
+  schedule:
+    # Runs "At 23:59 every night" (UTC)
+    - cron: '59 23 * * *'
+
+jobs:
+  fetch-changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Upload CHANGELOG.md
+        uses: actions/upload-artifact@v4
+        with:
+          name: changelog
+          path: CHANGELOG.md
+
+  create-new-release:
+    runs-on: ubuntu-latest
+    needs: fetch-changelog
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Download CHANGELOG.md
+        uses: actions/download-artifact@v4
+        with:
+          name: changelog
+      - name: Check if there are any changes in CHANGELOG.md
+        id: verify-diff
+        run: |
+          git diff --quiet . || echo "changed=true" >> $GITHUB_OUTPUT
+      - name: Parse CHANGELOG for today's entries and create a new tag
+        if: steps.verify-diff.outputs.changed == 'true'
+        run: |
+          # Extract the date of today from the CHANGELOG and filter relevant entries
+          TODAY=$(date -u +'%Y-%m-%d')
+          # Find the lines in the CHANGELOG corresponding to today’s date
+          LINES=$(nl -v0 -ba CHANGELOG.md | grep "## $TODAY" - | head -n 2 | awk '{print $1}')
+          DESCRIPTION=$(cat CHANGELOG.md | head -n $(echo $LINES | cut -d' ' -f2) | tac | head -n -$(echo $LINES | cut -d' ' -f1) | head -n -1 | tac)
+
+          # Git commit configuration
```

Comment:
this section should be removed

We only need to see if CHANGELOG section corresponding to today’s date exist, and if it does, use it as release body. No need to any pushing or git diffing